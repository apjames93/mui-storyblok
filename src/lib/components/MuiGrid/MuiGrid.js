import React from 'react';
import Grid from '../Grid/Grid';
import MuiAppBar from '../MuiAppBar/MuiAppBar';
import MuiMobileStepper from '../MuiMobileStepper/MuiMobileStepper';
import MuiHeroHeader from '../MuiHeroHeader/MuiHeroHeader';
import MuiCircularProgress from '../MuiCircularProgress/MuiCircularProgress';
import MuiGridList from '../MuiGridList/MuiGridList';
import MuiButtonRedirect from '../MuiButtonRedirect/MuiButtonRedirect';
import MuiCard from '../MuiCard/MuiCard';
import MuiTypography from '../MuiTypography/MuiTypography';
import MuiMenu from '../MuiMenu/MuiMenu';
import MuiIcon from '../MuiIcon/MuiIcon';
import MuiAccordion from '../MuiAccordion/MuiAccordion';
import MuiList from '../MuiList/MuiList';
import MuiTable from '../MuiTable/MuiTable';
import MuiPaginationTable from '../MuiPaginationTable/MuiPaginationTable';
import MuiContactButton from '../MuiContactButton/MuiContactButton';
import MuiImage from '../MuiImage/MuiImage';
import MuiButtonHref from '../MuiButtonHref/MuiButtonHref';
import MuiNotificationBanner from '../MuiNotificationBanner/MuiNotificationBanner';
import BlokForm from '../BlokForm/BlokForm';
import MuiAudioPlayer from '../MuiAudioPlayer/MuiAudioPlayer';
import GoogleConversions from '../GoogleConversions/GoogleConversions';
import MuiIconButtonDialog from '../MuiIconButtonDialog/MuiIconButtonDialog';
import MuiButtonDialog from '../MuiButtonDialog/MuiButtonDialog';
import MuiGeoLocationMobileStepper from '../MuiGeoLocationMobileStepper/MuiGeoLocationMobileStepper';

const MuiGrid = (props) => {
  const components = {
    MuiHeroHeader,
    MuiAppBar,
    MuiMobileStepper,
    MuiCircularProgress,
    MuiGridList,
    MuiButtonRedirect,
    MuiCard,
    MuiTypography,
    MuiMenu,
    MuiIcon,
    MuiAccordion,
    MuiList,
    MuiTable,
    MuiPaginationTable,
    MuiContactButton,
    MuiImage,
    MuiButtonHref,
    MuiNotificationBanner,
    BlokForm,
    MuiAudioPlayer,
    GoogleConversions,
    MuiGeoLocationMobileStepper,
    MuiIconButtonDialog,
    MuiButtonDialog,
  };

  return (
    <Grid
      {...props}
      components={components}
    />
  );
};

export default MuiGrid;
