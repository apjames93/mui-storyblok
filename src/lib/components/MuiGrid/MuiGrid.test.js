import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiGrid from './MuiGrid';

function setup() {
  const props = {
    _editable: '<!--#storyblok#{"name": "MuiGrid", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
    content: [{
      _editable: '<!--#storyblok#{"name": "MuiGridItem", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'MuiGridItem',
      content: [{
        _editable: '<!--#storyblok#{"name": "MuiIcon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
        component: 'MuiIcon',
        iconName: 'android',
      }],
    }],
  };
  const comp = shallow(<MuiGrid {...props} />);
  return { comp, props };
}

describe('<MuiGrid />', () => {
  it('renders MuiGrid', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiGrid {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
