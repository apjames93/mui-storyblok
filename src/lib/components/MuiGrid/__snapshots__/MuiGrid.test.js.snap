// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiGrid /> snapshot 1`] = `
<div
  className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    className="MuiGrid-root makeStyles-root-2 MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-item MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <div
      className="MuiIcon storyblok--outline"
      data-blok-c="{\\"name\\": \\"MuiIcon\\", \\"space\\": \\"48408\\", \\"uid\\": \\"947be9f0-47c3-4315-a95a-550f0c560eb5\\", \\"id\\": \\"307934\\"}"
    >
      <span
        aria-hidden={true}
        className="material-icons MuiIcon-root makeStyles-root-3 MuiIcon-colorSecondary"
      >
        android
      </span>
    </div>
  </div>
</div>
`;
